File: scripts/remove-bg.py
import os
from pathlib import Path
from rembg import remove
from PIL import Image

SRC = Path("public/images")
OUT = Path("public/images/nobg")
OUT.mkdir(parents=True, exist_ok=True)

EXTS = {".webp", ".png", ".jpg", ".jpeg"}
SKIP = {"gamers-end-logo.png", "provided-logo.png", "gamecraft-logo.webp", "product-sheet.png", "design-reference.png"}

images = [f for f in SRC.iterdir() if f.suffix.lower() in EXTS and f.name not in SKIP]
print(f"Processing {len(images)} images...")

for img_path in sorted(images):
    out_name = img_path.stem + ".png"
    out_path = OUT / out_name
    print(f"  {img_path.name} -> {out_name}", end=" ... ", flush=True)
    try:
        inp = Image.open(img_path)
        result = remove(inp)
        result.save(out_path, "PNG")
        print("done")
    except Exception as e:
        print(f"FAILED: {e}")

print("All done!")

File: scripts/remove-bg.mjs
import { removeBackground } from '@imgly/background-removal';
import { writeFile, mkdir } from 'fs/promises';
import { join, extname } from 'path';

const SRC = 'public/images';
const OUT = 'public/images/nobg';

const TARGETS = [
  'consoles.webp',
  'controllers.webp',
  'headsets.webp',
  'keyboards.webp',
  'mice.webp',
  'monitors.webp',
  'games.webp',
  'accessories.webp',
  'consoles-detail.webp',
  'controllers-detail.webp',
  'headsets-detail.webp',
  'keyboards-detail.webp',
  'mice-detail.webp',
  'monitors-detail.webp',
  'games-detail.webp',
  'accessories-detail.webp',
  'hero-controller.webp',
  'reference-console.webp',
];

async function main() {
  await mkdir(OUT, { recursive: true });

  console.log(`Processing ${TARGETS.length} images...`);

  for (const file of TARGETS) {
    const srcPath = join(SRC, file);
    const outName = file.replace(extname(file), '.png');
    const outPath = join(OUT, outName);

    console.log(`  ${file} → ${outName}`);
    try {
      const buffer = await removeBackground(srcPath, {
        model: 'isnet_fp16',
        output: { format: 'image/png', quality: 1 },
      });
      await writeFile(outPath, buffer);
      console.log(`    done`);
    } catch (err) {
      console.error(`    failed: ${err.message}`);
    }
  }
  console.log('All done!');
}

main();
